scripts/check_samples.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import os
+import pathlib
+import re
+import subprocess
+import sys
+import time
+
+
+def strip(text, suffix):
+    if suffix and text.endswith(suffix):
+        return text[:-len(suffix)]
+    return text
+
+def get_expected_samples(crd_path):
+    # For some reason it's 2-5x faster to do this with yq than it is to do it with pyyaml
+    result = subprocess.run(['yq', 'e', '{"name": .spec.names.singular, "group": .spec.group, "version": .spec.versions[] | select(.name != "*storage").name}', str(crd)], capture_output=True, text=True)
+
+    if result.returncode != 0:
+        print(f"yq failed ({result.returncode}): {result.stderr}")
+        sys.exit(1)
+
+    expected_samples = set()
+    lines = result.stdout.strip().splitlines()
+
+    # TODO: There's probably a way to do this in a list comprehension but my Python is clearly rusty
+    kind = None
+    group = None
+    version = None
+    for line in lines:
```

Comment:
So each gvk is on 3 lines? I'd use the `grouper` recipe on [this list](https://docs.python.org/3/library/itertools.html#itertools-recipes) and then iterate through them with `for kind, group, version in grouper(lines, 3):`.  

Or you could use `cut` to chop off the first part of the line and `paste` to stick them together in bash? But I guess you still have the checking below to do in Python.

Reply:
Good suggestion - done.

Reply:
For clarification, did the `grouper` bit.